tests/symlinked_included_files.test.ts

```diff
@@ -0,0 +1,65 @@
+import { readdir } from 'fs/promises'
+import { join } from 'path'
+
+import decompress from 'decompress'
+import { dir as getTmpDir } from 'tmp-promise'
+import { expect, test } from 'vitest'
+
+import { ARCHIVE_FORMAT, zipFunction } from '../src/main.js'
+
+import { FIXTURES_ESM_DIR } from './helpers/main.js'
+
+const readDirWithTypes = async (dir: string, readFiles?: Record<string, boolean>, parent = '') => {
```

Comment:
Should this be moved to a helper? Ideally with some JSDoc explaining what it does.